src/SIL.XForge.Scripture/Services/SFProjectService.cs

```diff
@@ -502,13 +511,35 @@ public async Task<string> GetLinkSharingKeyAsync(string curUserId, string projec
                     {
                         Key = key,
                         ProjectRole = role,
-                        ExpirationTime = null
+                        ExpirationTime = (shareLinkType == ShareLinkType.Recipient ? expTime : null),
+                        ShareLinkType = shareLinkType
                     }
                 )
         );
         return key;
     }
 
+    public async Task<bool> ReserveLinkSharingKeyAsync(string curUserId, string shareKey)
```

Comment:
This method doesn't seem to need a return type, as it only ever returns `true`? (see also my comment on `ISFProjectService.ReserveLinkSharingKeyAsync()`)